libraries/typings-generator/src/TypingsGenerator.ts

```diff
@@ -106,46 +110,56 @@ export class TypingsGenerator<TFileContents = string> {
   );
   public constructor(options: ITypingsGeneratorOptionsWithCustomReadFile<string | undefined, TFileContents>);
   public constructor(options: ITypingsGeneratorOptionsWithCustomReadFile<string | undefined, TFileContents>) {
-    this._options = {
+    this._options = options = {
       ...options,
       readFile:
         options.readFile ??
         ((filePath: string, relativePath: string): Promise<TFileContents> =>
-          FileSystem.readFileAsync(filePath) as Promise<TFileContents>)
+          FileSystem.readFileAsync(filePath) as Promise<TFileContents>),
+      writeFileAsync:
+        options.writeFileAsync ??
+        ((filePath: string, contents: string): Promise<void> =>
+          FileSystem.writeFileAsync(filePath, contents, {
+            ensureFolderExists: true,
+            convertLineEndings: NewlineKind.OsDefault
```

Comment:
Why are we doing newline conversion? We should really only ever be using a newline other than `\n` if writing shell scripts. Not a new change, though.